### PR TITLE
Project settings

### PR DESCRIPTION
**Pull request by microsoftsam**
Changing rename to project settings in "More" dropdown menu 
- adding save button in settings
- adding delete button in settings
- disabling auto save

screen shot 2017-01-10 at 2 51 19 pm

### CODE REVIEW

File: webapp/src/pxtjson.tsx [hunk 1]
```diff
@@ -68,12 +77,12 @@ export class Editor extends srceditor.Editor {
...
             }
             // trigger update            
-            update(uc);
+            save();
         }
         return (
             <div className="ui content">
                 <div className="ui segment form text" style={{ backgroundColor: "white" }}>
-                    <sui.Input label={lf("Name")} value={c.name} onChange={v => update(c.name = v)} />
+                    <sui.Input label={lf("Name")} value={c.name} onChange={(v) => updateName(c.name = v)} />
```
Comment (pelikhan) [inline webapp/src/pxtjson.tsx]: we should not use the onChange, it is broken for script languages. Can we do without it?

File: webapp/src/pxtjson.tsx [hunk 2]
```diff
@@ -82,7 +91,9 @@ export class Editor extends srceditor.Editor {
                             onChange={() => applyUserConfig(uc) } />
                     ) }
                     <sui.Field>
+                        <sui.Button text={lf("Save")} class={`green ${this.isSaving ? 'disabled' : ''}`} onClick={() => save()} />
                         <sui.Button text={lf("Edit Settings As text") } onClick={() => this.parent.editText() } />
+                        <sui.Button text={lf("Delete")} class={`red right floated`} onClick={() => deleteProject()} />
```
Comment (pelikhan) [inline webapp/src/pxtjson.tsx]: We do have a "delete project" button in the menu, why are we adding it again?

File: webapp/src/pxtjson.tsx [hunk 3]
```diff
@@ -23,15 +24,23 @@ export class Editor extends srceditor.Editor {
...
+        }
+        const save = () => {
+            this.isSaving = true;
             const f = pkg.mainEditorPkg().lookupFile("this/" + pxt.CONFIG_NAME);
             f.setContentAsync(JSON.stringify(this.config, null, 4) + "\n").then(() => {
                 pkg.mainPkg.config.name = c.name;
                 this.parent.setState({projectName: c.name});
                 this.parent.forceUpdate()
                 Util.nextTick(this.changeCallback)
+                this.isSaving = false;
```
Comment (pelikhan) [inline webapp/src/pxtjson.tsx]: when done saving, we should reload the project

File: webapp/src/pxtjson.tsx (hunk 1 above)
Comment (microsoftsam) [inline webapp/src/pxtjson.tsx]: I think we can use ref, and extract the value of the input field on save.

File: webapp/src/pxtjson.tsx (hunk 2 above)
Comment (microsoftsam) [inline webapp/src/pxtjson.tsx]: Thought it made sense in the Settings page. Up for discussion.

Comment (pelikhan) [inline webapp/src/pxtjson.tsx]: We should avoid having multiple ways of doing the same thing. IMO.

Comment (abchatra) [inline webapp/src/pxtjson.tsx]: May be remove from the menu? Delete makes more sense Project settings view.

Comment (pelikhan) [inline webapp/src/pxtjson.tsx]: Really hard to discover where the "delete project" button is when only in settings.

Comment (microsoftsam) [inline webapp/src/pxtjson.tsx]: Removing from settings.

**Comment by microsoftsam** [comment]
Fixes #542

**Comment by microsoftsam** [comment]
screen shot 2017-01-13 at 10 21 03 am

**Pull request merged**